tool/tsh/db.go

```diff
@@ -1119,6 +1119,20 @@ func withConnectRequirements(tc *client.TeleportClient, route *tlsca.RouteToData
 			// (elasticsearch-sql-cli cannot be configured to use specific certs).
 			r.requireLocalProxyWithTunnel(formatDBProtocolReason(route.Protocol))
 		}
+		if r.localProxy && r.tunnel {
+			return
+		}
+		// Call API and check if a user needs to use MFA to connect to the database.
+		mfaRequired, err := isMFADatabaseAccessRequired(ctx, tc, route)
+		if err != nil {
+			log.WithError(err).Debugf("error getting MFA requirement for database %v",
+				route.ServiceName)
+		} else if mfaRequired {
+			// When MFA is required, we should require a local proxy tunnel,
+			// because the local proxy tunnel can hold database MFA certs in-memory
+			// without a restricted 1-minute TTL. This is better for user experience.
+			r.requireLocalProxyWithTunnel("MFA is required to connect to the database")
```

Comment:
The `requireLocalProxyWithTunnel` don't have any effect by the require... naming suggest the this function will trigger some kind of validation. What would you say about renaming it to something with  `Add...` prefix ?

Reply:
sure we can rename the prefix